From 9713c36a41195984919937e7b8f0c5f80a151f85 Mon Sep 17 00:00:00 2001
From: Finley Xiao <finley.xiao@rock-chips.com>
Date: Tue, 2 Jun 2020 14:11:30 +0800
Subject: [PATCH] ARM: dts: rv1126: set dummy_cpll as fixed clk

If the parent of one clock is cpll and we set it as dummy clock, the
clock will be a orphan clock and the clk provider will register abortively,
as follows.

[    0.000000] rockchip_clk_of_add_provider: could not register clk provider

The cause of the error:
struct clk *__clk_create_clk(struct clk_hw *hw, const char *dev_id,
                             const char *con_id, bool with_orphans)
{
        /* This is to allow this function to be chained to others */
        if (!hw || IS_ERR(hw))
                return (struct clk *) hw;

        if (hw->core->orphan && !with_orphans)
                return ERR_PTR(-EPROBE_DEFER);

        return clk_hw_create_clk(hw, dev_id, con_id);
}

Signed-off-by: Finley Xiao <finley.xiao@rock-chips.com>
Change-Id: I7053cd4e0a028a1556c774601d5c01578ee5b1ac
---
 arch/arm/boot/dts/rv1126.dtsi | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/arm/boot/dts/rv1126.dtsi b/arch/arm/boot/dts/rv1126.dtsi
index 851c717a941d..f96cd08e23bf 100644
--- a/arch/arm/boot/dts/rv1126.dtsi
+++ b/arch/arm/boot/dts/rv1126.dtsi
@@ -369,6 +369,13 @@
 		#clock-cells = <0>;
 	};
 
+	dummy_cpll: dummy_cpll {
+		compatible = "fixed-clock";
+		clock-frequency = <0>;
+		clock-output-names = "dummy_cpll";
+		#clock-cells = <0>;
+	};
+
 	gmac_clkin_m0: external-gmac-clockm0 {
 		compatible = "fixed-clock";
 		clock-frequency = <125000000>;
-- 
2.35.3

